glean-core/src/metrics/mod.rs

```diff
@@ -312,3 +339,34 @@ impl Metric {
         }
     }
 }
+
+macro_rules! impl_malloc_size_of_for_metric {
+    ($ty:ident) => {
+        impl ::malloc_size_of::MallocSizeOf for $ty {
+            fn size_of(&self, ops: &mut malloc_size_of::MallocSizeOfOps) -> usize {
+                // Note: This is behind an `Arc`.
+                // `size_of` should only be called on the main thread to avoid double-counting.
+                self.meta().size_of(ops)
+            }
+        }
+    };
+}
+
+impl_malloc_size_of_for_metric!(BooleanMetric);
+impl_malloc_size_of_for_metric!(CounterMetric);
+impl_malloc_size_of_for_metric!(CustomDistributionMetric);
```

Comment:
So we're just gonna ignore `range_min`, `range_max`, `bucket_count`, and `histogram_type`?

Also, why does this need to be a macro? Can't we impl it over all T that are `impl MetricType`? (I guess that could conflict with the specialized versions... but we could put the default behaviour in the trait and override it in the metrictype's impl, maybe?

Reply:
None of the parameters you mentioned allocate.

For why it cannot be a generic impl:

```
343 | impl<T: MetricType> ::malloc_size_of::MallocSizeOf for T where T: MetricType {
    |      ^ type parameter `T` must be used as the type parameter for some local type
```